### PR TITLE
Decompose to native_dropout in eval mode as well

### PR DESCRIPTION
**Pull request by andrewor14**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #113492
* __->__ #109914

Summary: https://github.com/pytorch/pytorch/pull/106274 added a
decomp for dropout such that for train mode we get
`aten.native_dropout`, while for eval mode we get `aten.clone`.
This commit makes it such that we get `aten.native_dropout` for
both train and eval modes. For eval mode, we let the cloning
happen in the aten op itself.

The main motivation behind this change is QAT, which needs to
swap between `aten.native_dropout(train=True)` and
`aten.native_dropout(train=False)` in the graph. This was
previously difficult to do since there was no dropout op to
match and replace in eval mode.

Test Plan: python test/test_ops.py

Reviewers: SherlockNoMad, bdhirsh

Subscribers: SherlockNoMad, bdhirsh, supriyar

cc @avikchaudhuri @gmagogsfm @zhxchen17 @tugsbayasgalan @angelayi @suo @ydwu4

### CODE REVIEW

**Review by jerryzh168 — APPROVED** [review]
LG, thanks!

**Comment by andrewor14** [comment]
Closing this in favor of the alignment here: https://github.com/pytorch/pytorch/pull/115258#issuecomment-1900755832
